amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/json/JSONTokenerHack.scala

```diff
@@ -34,66 +33,28 @@ class JSONTokenerHack(text: String) extends JSONTokener(text) {
 
         val string = sb.toString.trim()
         if ("" == string) throw this.syntaxError("Missing value")
-        stringToValue(string)
+        JSONObject.stringToValue(string)
     }
   }
 
   protected def shouldContinueParsing(newChar: Char) = newChar >= ' ' && ",:]}/\\\"[{;=#".indexOf(newChar) < 0
 
-  private def checkNumber(s: String): Option[Object] =
-    try {
-      if (isDecimalNotation(s) || isBiggerThanLong(s)) checkDouble(s) else checkLong(s)
-    } catch {
-      case _: Exception => None
+  private def hack(value: Object) =
```

Comment:
Add a more descriptive method name and a description if possible (why we do this hack)

Reply:
adjusted 👍